### PR TITLE
[kodi] Add channels for screensaver state and virtual keyboard state and include OnAVStart for playing state detection

### PR DESCRIPTION
**Pull request by Flole998**
Fixes: #7542

### CODE REVIEW

File: bundles/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/KodiEventListener.java
Comment (lolodomo) [inline bundles/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/KodiEventListener.java:54]: Please also fix the parameter here

File: bundles/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java
Comment (lolodomo) [inline bundles/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java:71]: Why using this new kind of naming for property?

File: bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml
Comment (lolodomo) [inline bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml:565]: Please also update the default translation properties

**Review by lolodomo — CHANGES_REQUESTED** [review]

**Comment by lolodomo** [comment]
@pail23 and @cweitkamp : would you like to review too?

File: bundles/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java
Comment (Flole998) [reply bundles/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java:71]: That's simply how Kodi is calling it.

File: bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml
Comment (Flole998) [reply bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml:565]: Fixed

File: bundles/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/KodiEventListener.java
Comment (Flole998) [reply bundles/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/KodiEventListener.java]: Fixed

File: bundles/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java
Comment (lolodomo) [reply bundles/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java:71]: It is not a thing property?

Comment (Flole998) [reply bundles/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java:71]: No, all these `PROPERTY_*` are properties of the JSON that is sent to/from Kodi.

Comment (lolodomo) [reply bundles/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java:71]: Ok, thank you

**Review by lolodomo — APPROVED** [review]
Code LGTM

**Comment by lolodomo** [comment]
I let a chance to binding maintainers to review before merging. 
@pail23 @cweitkamp

File: bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml
Comment (cweitkamp) [inline bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml:571]: I propose to model this as a trigger channel which emits the requested input type (or "FINISHED"). This way a user can implement different reactions based on the requested type. Wdyt?

File: bundles/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java
Comment (cweitkamp) [inline bundles/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java:1000]: Nice finding 👍🏼 .

**Review by cweitkamp — CHANGES_REQUESTED** [review]
Thanks for this improvement. I am missing the documentation for this new feature.

And I left a comment regarding the modelling of the `inputrequested` channel.

File: bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml
Comment (Flole998) [reply bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml:571]: I'm trying to get a change into Kodi which allows detecting if an Input is requested or not. Currently, if openHAB is started up after the input is requested it wouldn't notice it, even though the dialog is still open. The idea was to add a boolean for that, so that it could be implemented similar to the screensaver detection. If this would indeed be modeled as a trigger channel this future enhancement would be a lot harder.

Also we actually have "Title" (like "Enter search String") and the type, so ideally we would give access to both. If you still think it's better to make it a trigger channel (with the disadvantage that at least I won't be able to implement the polling on startup/reconnection) then it might make more sense to use the title and not the type.

**Comment by lolodomo** [comment]
Oh yes, I forgot the missing update of the documentation. Hopefully I did not merge it.

File: bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml
Comment (cweitkamp) [reply bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml:571]: This will not work either. The `Input.OnInputRequested` is a Kodi notification which will be triggered only once when the user input is necessary. Thus if openHAB is not available at that specific moment or the websocket of the binding is not connected it will never be recognized by OH or set after a restart. You are not polling for it. Even more the state of the item linked to the channel will be `NULL` after an restart.

Regarding the title I can think of an OH enhancement to define additional properties delivered by an event. Currently an event can only have a type no other payload.

Comment (Flole998) [reply bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml:571]: Exactly, there is currently no way of polling it in Kodi. But I am working on implementing that, it will just be a boolean though.

Comment (cweitkamp) [reply bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml:571]: Alright. Please ping me once you got it implemented.

File: bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml
Comment (jlaur) [inline bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml:568]: These texts (and inputrequested) must be added to resources/OH-INF/i18n/kodi.properties as well:
```
channel-type.kodi.screensaver.label = Screensaver
channel-type.kodi.screensaver.description= Status of the screensaver
channel-type.kodi.inputrequested.label = InputRequested
channel-type.kodi.inputrequested.description= Status that shows if Kodi is asking for user input.
```

File: bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml
Comment (jlaur) [inline bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml:573]: "Input requested" in two words?

File: bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml
Comment (Flole998) [reply bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml:571]: Just to confirm: You want to wait for the change before merging? That change needs to be done at Kodi first and the maintainers there have already made clear on a different PR that there won't be API-changes between versions, so before it could actually be used a new version needs to be released with those changes in. I don't think waiting for that is a good idea as that could very well take another year or so, so I'd rather kick out this change if it's a reason not to merge and put it in a diferent PR.

File: bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml
Comment (Flole998) [reply bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml:568]: It's already there, isn't it?

File: bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml
Comment (Flole998) [reply bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml:573]: Yes, that's probably better. Thanks!

File: bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml
Comment (jlaur) [reply bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml:568]: Right. I somehow missed the file, sorry.

File: bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml
Comment (Flole998) [reply bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml]: Fixed

**Comment by Flole998** [comment]
I just updated the README.md aswell.

**Comment by lolodomo** [comment]
@cweitkamp  and @jlaur : please have the lead to finish the review & merge of this PR.

File: bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml
Comment (jlaur) [reply bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml:571]: @cweitkamp, @Flole998 - can this discussion be concluded? Anything preventing this PR for being merged?

Comment (cweitkamp) [reply bundles/org.openhab.binding.kodi/src/main/resources/OH-INF/thing/thing-types.xml:571]: Yes, I am fine with it. We can merge this now.

**Review by cweitkamp — APPROVED** [review]

**Review by jlaur — APPROVED** [review]
Thanks for this contribution!

**Pull request merged**
